grails-app/assets/javascripts/streama/controllers/admin-show-ctrl.js

```diff
@@ -84,29 +86,62 @@ angular.module('streama').controller('adminShowCtrl', [
 	};
 
 	var seasonForShow = function (season) {
-		apiService.theMovieDb.seasonForShow({apiId: $scope.show.apiId, showId: $stateParams.showId, season: season})
+		return apiService.theMovieDb.seasonForShow({apiId: $scope.show.apiId, showId: $stateParams.showId, season: season})
 				.success(function (data) {
 					console.log('%c seasonForShow', 'color: deeppink; font-weight: bold; text-shadow: 0 0 5px deeppink;');
 					$scope.seasons = $scope.seasons ||  {};
 					$scope.seasons[season] = $scope.seasons[season] || [];
 					$scope.seasons[season] = $scope.seasons[season].concat(data);
-					$scope.loading = false;
 					$scope.newEpisodesForSeason = null;
-					$scope.setCurrentSeason(season);
-					alertify.success( data.length + ' Episodes fetched');
-				}).error(function () {
+					maxSeason = Math.max(maxSeason, season);
+					if (maxSeason == season) {
+						$scope.setCurrentSeason(season);
+					}
+					episodesFetched += data.length;
+				});
+	};
+
+	var getEpisodesForSeasons = function (seasons) {
+		var promises = [];
+		for (var i = 0; i < seasons.length; i++) {
+			promises.push(seasonForShow(seasons[i]));
+		}
+
+		$q.all(promises).then(function() {
 			$scope.loading = false;
+			maxSeason = 0;
+			alertify.success(episodesFetched + " Episodes fetched");
+			episodesFetched = 0;
 		});
 	};
 
+	$scope.fetchEpisodes = function(){
+    	alertify.set({ buttonReverse: true, labels: {ok: "OK", cancel : "Cancel"}});
+		alertify.prompt("For which seasons would you like to fetch the episodes? (Leave blank to fetch for all seasons)", function (confirmed, season) {
```

Comment:
maybe give user more clear instructions, for example that he can use 1-3 syntax.